### PR TITLE
[WIP][AIRFLOW-4534] Remove redundant list around dict comprehensions

### PR DESCRIPTION
**Pull request by BasPH**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4534
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Following up on https://github.com/apache/airflow/pull/5295, there are several redundant `list()` calls around dict comprehensions. This wastes time and is therefore removed in this PR.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by BasPH** [comment]
Hmm turns out this is more difficult than expected.

Looking specifically into `_update_counters()` now. `list(ti_status.running.items())` creates a copy of the dict, and this copy is not edited while the dict itself is edited while running through the for loop. When iterating over just the dict you receive a RuntimeError:

```
RuntimeError: dictionary changed size during iteration
```

While the list is a copy which isn't edited at runtime, so that's fine to iterate over. Unsure if this should be expected behaviour or if the whole thing is borked. Setting to WIP for now.

**Comment by andriisoldatenko** [comment]
@BasPH yes, it's known behaviour for any mutable collections:
```
In [1]: d = {'a': 1, 'b': 2}

In [2]: for k,v in d.items():
   ...:     d['x'] = 1
   ...:
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-cdc4495c7b4d> in <module>
----> 1 for k,v in d.items():
      2     d['x'] = 1
      3

RuntimeError: dictionary changed size during iteration
```

**Review by ashb — APPROVED** [review]

File: airflow/jobs/backfill_job.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def _update_counters(self, ti_status):
         :param ti_status: the internal status of the backfill job tasks
         :type ti_status: BackfillJob._DagRunTaskStatus
         """
-        for key, ti in list(ti_status.running.items()):
+        for key, ti in ti_status.running.items():
```
Comment (ashb) [inline airflow/jobs/backfill_job.py:188]: This one at least is still needed:

```
    airflow/jobs/backfill_job.py line 188 in _update_counters
      for key, ti in ti_status.running.items():
   RuntimeError: dictionary changed size during iteration
```

**Comment by BasPH** [comment]
not stale

**Comment by dimberman** [comment]
@BasPH I'm going to close this but please feel free to re-open if you feel this is still relevant.

**Pull request closed without merging**